fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/info/TableNameInfo.java

```diff
@@ -0,0 +1,168 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+// This file is copied from
+// https://github.com/apache/impala/blob/branch-2.9.0/fe/src/main/java/org/apache/impala/TableName.java
+// and modified by Doris
+
+package org.apache.doris.nereids.trees.plans.commands.info;
+
+import org.apache.doris.analysis.TableName;
+import org.apache.doris.catalog.Env;
+import org.apache.doris.cluster.ClusterNamespace;
+import org.apache.doris.common.io.Text;
+import org.apache.doris.common.io.Writable;
+import org.apache.doris.datasource.InternalCatalog;
+import org.apache.doris.nereids.exceptions.AnalysisException;
+import org.apache.doris.persist.gson.GsonUtils;
+import org.apache.doris.qe.ConnectContext;
+
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+import com.google.gson.annotations.SerializedName;
+
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * table name info
+ */
+public class TableNameInfo implements Writable {
+    @SerializedName(value = "c")
+    private String ctl;
+    @SerializedName(value = "t")
+    private String tbl;
+    @SerializedName(value = "d")
+    private String db;
+
+    public TableNameInfo() {
+
+    }
+
+    /**
+     * TableNameInfo
+     * @param parts like [ctl1,db1,tbl1] or [db1,tbl1] or [tbl1]
+     */
+    public TableNameInfo(List<String> parts) {
+        Objects.requireNonNull(parts, "require parts object");
```

Comment:
Extract catalgot, db, table name from List<String> should be a common method, Maybe something like `org.apache.doris.nereids.util.RelationUtil#getQualifierName` should be in common util

Reply:
Is it also possible in the construction method?just like `org.apache.doris.analysis.TableName`